go/workflow.go

```diff
@@ -0,0 +1,333 @@
+	activities := Activities{}
+
+	var pointsToAdd int
+	c.Receive(ctx, &pointsToAdd)
+
+	logger.Info("Adding points to customer account.", "PointsAdded", pointsToAdd)
+	customer.LoyaltyPoints += pointsToAdd
+
+	currentStatusOrd := customer.StatusLevel.Ordinal
+	customer.StatusLevel = StatusLevelForPoints(customer.LoyaltyPoints)
+	newStatusOrd := customer.StatusLevel.Ordinal
+
+	statusChange := newStatusOrd - currentStatusOrd
+
+	if statusChange > 0 {
+		err := workflow.ExecuteActivity(ctx, activities.SendEmail,
+			fmt.Sprintf(EmailPromoted, customer.StatusLevel.Name)).
+			Get(ctx, nil)
+		if err != nil {
+			return fmt.Errorf("error running SendEmail activity for status promotion: %w", err)
+		}
+	} else if statusChange < 0 {
+		err := workflow.ExecuteActivity(ctx, activities.SendEmail,
+			fmt.Sprintf(EmailDemoted, customer.StatusLevel.Name)).
+			Get(ctx, nil)
+		if err != nil {
+			return fmt.Errorf("error running SendEmail activity for status demotion: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func signalInviteGuest(ctx workflow.Context, c workflow.ReceiveChannel, customer *CustomerInfo) error {
+	logger := workflow.GetLogger(ctx)
+	activities := Activities{}
+
+	var emailToSend string
+	var guestID string
+	c.Receive(ctx, &guestID)
+
+	logger.Info("Checking to see if customer has enough status to allow for a guest invite.", "Customer", customer)
+	nonEmptyGuests := 0
+	for _, v := range customer.Guests {
+		if len(v) > 0 {
+			nonEmptyGuests++
+		}
+	}
+	if nonEmptyGuests < customer.StatusLevel.GuestsAllowed {
+		logger.Info("Customer is allowed to invite guests. Attempting to invite.",
+			"GuestID", guestID)
+
+		guest := CustomerInfo{
+			CustomerID:    guestID,
+			StatusLevel:   customer.StatusLevel.Previous(),
+			AccountActive: true,
+		}
+
+		err := addGuestToCustomer(customer, guestID)
+		if err != nil {
+			return fmt.Errorf("unable to add guest '%v' to customer's guest list: %w", guestID, err)
+		}
+
+		err = workflow.ExecuteLocalActivity(ctx, activities.StartGuestWorkflow, guest).Get(ctx, nil)
+
+		t := &GuestAlreadyCanceledError{}
+		if errors.As(err, &t) {
+			emailToSend = EmailGuestCanceled
+		} else if err != nil {
+			return fmt.Errorf("could not signal-with-start guest/child workflow for guest ID '%v': %w", guestID, err)
+		} else {
+			emailToSend = EmailGuestInvited
+		}
+	} else {
+		logger.Info("Customer does not have sufficient status to invite more guests.")
+		emailToSend = EmailInsufficientPoints
+	}
+
+	err := workflow.ExecuteActivity(ctx, activities.SendEmail, emailToSend).Get(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("error running SendEmail activity: %w", err)
+	}
+
+	return nil
+}
+
+func signalEnsureMinimumStatus(ctx workflow.Context, c workflow.ReceiveChannel, customer *CustomerInfo) error {
+	activities := Activities{}
+
+	var minStatus StatusLevel
+	c.Receive(ctx, &minStatus)
+
+	minOrd := minStatus.Ordinal
+	currentOrd := customer.StatusLevel.Ordinal
+
+	if currentOrd < minOrd {
```

Comment:
```suggestion
	if customer.StatusLevel.Ordinal < minStatus.Ordinal {
```

Often it's just bloat to overly assign local variables for a single use